sktime/transformations/series/detrend/tests/test_mstl.py

```diff
@@ -0,0 +1,60 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Tests MSTL functionality."""
+
+__author__ = ["krishna-t"]
+
+import pytest
+import statsmodels as statsmodels
+
+from sktime.datasets import load_airline
+from sktime.transformations.series.detrend.mstl import MSTL
+
+
+@pytest.mark.skipif(
+    statsmodels.tsa.seasonal.MSTL is None, reason="MSTL is not installed"
+)
```

Comment:
Not sure if that's "better"? Mine is a shorter piece of code, so if mine works then it should be fine unless there's a deeper issue. Let's see what @fkiraly says. Thanks for the suggestion.

Reply:
There actually is a deeper issue.

We want to skip the test case when `statsmodels` is not installed. And we will have to change the implementation because of the following reasons
1. If `statsmodels` is not installed, the current implementation will raise an error instead of skipping the test.
2. All test cases that involve this scenario use `run_test_for_class` to check soft dependency, so we should keep the implementation coherent where possible.

Reply:
Ah, I see. That makes sense, thanks @geetu040. Though I wonder why the current implementation would raise an error rather than skip the test? Given that  the decorator specially is a `skipif` one. 

Does this have to do with the imports?

Reply:
> I presume this has something to do with the imports?

Yes you are right, import is the reason

https://github.com/sktime/sktime/pull/6084#issuecomment-1988097443
> The reason for this is because you are using `statsmodels.tsa.seasonal.MSTL` whereas the `MSTL` is not accessible like this, you need to import it separately from the module
> 
> ```python
> from statsmodels.tsa.seasonal import MSTL
> ```
> But this information is irrelevant as `MSTL` from `statsmodels` import is not needed when we are using `run_test_for_class` in `pytest.mark.skipif`